src/EnergyPlus/ElectricPowerServiceManager.hh

```diff
@@ -0,0 +1,1036 @@
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+// You are under no obligation whatsoever to provide any bug fixes, patches, or upgrades to the
+// features, functionality or performance of the source code ("Enhancements") to anyone; however,
+// if you choose to make your Enhancements available either publicly, or directly to Lawrence
+// Berkeley National Laboratory, without imposing a separate written license agreement for such
+// Enhancements, then you hereby grant the following license: a non-exclusive, royalty-free
+// perpetual license to install, use, modify, prepare derivative works, incorporate into other
+// computer software, distribute, and sublicense such enhancements or derivative works thereof,
+// in binary and source code form.
+
+#ifndef ElectricPowerServiceManager_hh_INCLUDED
+#define ElectricPowerServiceManager_hh_INCLUDED
+
+// C++ Headers
+#include <string>
+#include <vector>
+#include <memory>
+
+// ObjexxFCL Headers
+//#include <ObjexxFCL/Array1.hh>
+
+// EnergyPlus Headers
+#include <EnergyPlus.hh>
+#include <PlantLocation.hh>
+#include <OutputProcessor.hh>
+#include <DataHeatBalance.hh>
+#include <EMSManager.hh>
+
+namespace EnergyPlus {
+
+namespace ElectricPowerService {
+
+	enum thermalLossDestinationEnum {
+		heatLossNotDetermined,
+		zoneGains,
+		lostToOutside
+	};
+
+	void
+	initializeElectricPowerServiceZoneGains();
+
+
+
+class DCtoACInverter
+{
+private: // Creation
+	// Default Constructor
+		DCtoACInverter() :
+			aCPowerOut( 0.0 ),
+			aCEnergyOut( 0.0 ),
+			efficiency( 0.0 ),
+			dCPowerIn( 0.0 ),
+			dCEnergyIn( 0.0 ),
+			thermLossRate( 0.0 ),
+			thermLossEnergy( 0.0 ),
+			qdotConvZone( 0.0 ),
+			qdotRadZone( 0.0 ),
+			ancillACuseRate( 0.0 ),
+			ancillACuseEnergy( 0.0 ),
+			name( ""),
+			modelType( notYetSet ),
+			availSchedPtr( 0 ),
+			heatLossesDestination( heatLossNotDetermined ),
+			zoneNum( 0 ),
+			zoneRadFract( 0.0 ),
+			nightTareLossPower( 0.0 ),
+			nominalVoltage( 0.0 ),
+			nomVoltEfficiencyARR( 6, 0.0 ),
+			curveNum( 0 ),
+			ratedPower( 0.0 ),
+			minPower( 0.0 ),
+			maxPower( 0.0 ),
+			minEfficiency( 0.0 ),
+			maxEfficiency( 0.0 ),
+			standbyPower( 0.0 )
+
+		{}
+
+	// Copy Constructor
+	DCtoACInverter( DCtoACInverter const & ) = default;
+
+	// Move Constructor
+#if !defined(_MSC_VER) || defined(__INTEL_COMPILER) || (_MSC_VER>=1900)
```

Comment:
[Visual C++ 2013 doesn't support marking move constructors (or move assignment) as `= default`](https://msdn.microsoft.com/en-us/library/dn457344.aspx). This is fixed in VC++ 2015. It is not related to unique_ptr.

The reason for having them declared `= default` here is to make them private -- if you leave that out they are public. As to why this class has private construction I'm not sure: if it is meant to be a base class the constructors should be protected, not private, and the destructors should be virtual and assignment operators would normally be protected. If this is for a Named Constructor usage (unlikely) then where are the named constructors. I think some C++ review is needed here once the intent is clear.

In the sample AirTerminalUnit code I provided to @EnergyArchmage this `= default` usage was found in a protected section of a proper base class, which is not what we have here.